### PR TITLE
Fix testing methodology; passes accumulate now.

### PR DESCRIPTION
**Pull request by lavalamp**
`go test -count=10000 .` passes now.

I spot checked by introducing errors and the new tests continue to catch bugs. Running 10k tests as above caught the one place I messed up the fail/pass logic.

### CODE REVIEW

**Comment by lavalamp** [comment]
@dnephin @punkeel Would you like to review?

**Comment by lavalamp** [comment]
Added a commit to make all the tests work the same way, which has been bugging me since I wrote this library.

File: fuzz_test.go [hunk 1]
```diff
@@ -45,112 +46,130 @@ func TestFuzz_basic(t *testing.T) {
...
+	}
+	if s.mask != s2.mask {
+		panic("coding error, stages are differently typed")
+	}
+	s.passed |= (s2.passed & ^s2.failed)
+	return s
+}
+
+func (s Stages) AllPassed() bool {
+	return (s.passed & ^s.failed) == s.mask
```
Comment (dnephin) [inline fuzz_test.go:151]: Do you need to track failed if you have the mask? Wouldn't `s.passed == s.mask`  tell you if every stage passed at least once?  Maybe I'm not fully understanding all of these binary operations.

**Review by dnephin — COMMENTED** [review]
Nice! The new methodology looks great, and I like that a single one can be used for all the tests.

One question about tracking failures.

File: fuzz_test.go (hunk 1 above)
Comment (lavalamp) [reply fuzz_test.go:151]: Yeah, some of the code I replaced failed if any loop iteration failed. So it requires at least one pass and no failures to reproduce the prior logic. The tests could probably be adjusted to make this simpler, but that can happen later :)

File: fuzz_test.go [hunk 2]
```diff
@@ -45,112 +46,130 @@ func TestFuzz_basic(t *testing.T) {
...
+	failed uint64
+}
+
+func DeclareStages(N uint) Stages {
+	if N >= 63 {
+		panic("the math below only works up to 63")
+	}
+	return Stages{
+		// Exercise for the reader: make this work for all 64 bits.
+		mask: (uint64(1) << N) - 1,
```
[the remark this answers is not in the document]
Comment (lavalamp) [reply fuzz_test.go:119]: I may have done a little too much C / C++ early on in life :)

**Comment by lavalamp** [comment]
Thanks for looking, I'm gonna call that good enough, I can send a follow up if anyone finds a bug. :)

**Pull request merged**